Content.Server/Health/BodySystem/BodyManagerComponent.cs

```diff
@@ -355,5 +373,24 @@ public IEntity DropBodyPart(BodyPart part)
             }
         }
 
+        private void BodyPartAdded(BodyPart part, string slotName)
+        {
+            var argsAdded = new BodyPartAddedEventArgs(part, slotName);
+
+            foreach (var component in Owner.GetAllComponents<IBodyPartAdded>().ToArray())
+            {
+                component.BodyPartAdded(argsAdded);
+            }
+        }
+
+        private void BodyPartRemoved(BodyPart part, string slotName)
+        {
+            var args = new BodyPartRemovedEventArgs(part, slotName);
+
+            foreach (var component in Owner.GetAllComponents<IBodyPartRemoved>())
+            {
+                component.BodyPartRemoved(args);
+            }
+        }
```

Comment:
Couldn't this all simply be two events?

Reply:
Yes but BodySystem loads the body parts in ExposeData right now, making it very hard to hook into if it was events.